impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java

```diff
@@ -266,14 +271,14 @@ public String getBootstrapControllers() {
             throw new UnsupportedOperationException("Zookeeper based clusters don't support this operation");
         }
 
-        return buildBrokerList(nodeId -> getEndpointPair(Listener.CONTROLLER, nodeId));
+        return buildAdvertisedServers(this::isController, Listener.CONTROLLER);
     }
 
-    private synchronized String buildBrokerList(Function<Integer, KafkaClusterConfig.KafkaEndpoints.EndpointPair> endpointFunc) {
+    private synchronized String buildAdvertisedServers(Predicate<Integer> nodePredicate, Listener listener) {
```

Comment:
does the predicate actually need to be passed in?

Reply:
advertised servers can be the list of nodes in the KRaft role, or (currently) all nodes, so I think the predicate is important.